vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go

```diff
@@ -128,11 +142,39 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 	}
 	if exist {
 		glog.Infof("reconciling machine object %v triggers idempotent update.", name)
-		return c.update(m)
+		err = c.update(m)
+		if err != nil {
+			requeueErr, ok := err.(*RequeueAfterError)
+			if ok {
+				glog.Infof("actuator returned requeue after error: %v", requeueErr)
+				c.enqueueAfter(machine, requeueErr.RequeueAfter)
+				return nil
+			}
+		}
+		return err
 	}
 	// Machine resource created. Machine does not yet exist.
 	glog.Infof("reconciling machine object %v triggers idempotent create.", m.ObjectMeta.Name)
-	return c.create(m)
+	err = c.create(m)
+	if err != nil {
+		requeueErr, ok := err.(*RequeueAfterError)
+		if ok {
+			glog.Infof("actuator returned requeue after error: %v", requeueErr)
+			c.enqueueAfter(machine, requeueErr.RequeueAfter)
+			return nil
+		}
+	}
+	return err
+}
+
+func (c *MachineControllerImpl) enqueueAfter(machine *clusterv1.Machine, after time.Duration) error {
+	key, err := cache.MetaNamespaceKeyFunc(machine)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", machine, after)
+		return err
+	}
+	c.informers.WorkerQueues["Machine"].Queue.AddAfter(key, after)
```

Comment:
Why not just store the queue in an instance variable instead of the informers?

Reply:
Not sure if I got this to what you had in mind. Direct access to the queue is up in the generated controller code, and I'm not clear if that is permanently regenerated code, or was a one time generation. Will see what happens in upstream review.

Reply:
What I meant that instead of having an instance variable `c.informers`, you could have one named `c.queue` that gets initialized to informers.WorkerQueues["Machine"].Queue at the beginning.

Reply:
This didn't work out, the worker queue doesn't appear to exist when the controller is Init'd, so had to flip back to this original approach. Pursuing that fix upstream now.

Reply:
No worries I should have tested it better. Not sure if I didn't test it at all, or missed that the pod crashed and restarted, effectively resyncing it and looking like it worked. :)